Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift

```diff
@@ -33,28 +33,204 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
+                    switch result {
+                    case let .success(image):
+                        avatars.append(image)
+                    case let .failure(failure):
+                        avatars.append(placeholders[urlIndex])
+                        log.error(failure.localizedDescription)
+                    }
+                    group.leave()
+                }
+            } else {
+                avatars.append(appearance.images.avatarPlaceholders[urlIndex])
             }
-
-            // TODO: https://stream-io.atlassian.net/browse/CIS-652
-            // this is just placeholder implementation:
-
-            let firstOtherMember = channel.lastActiveMembers
-                .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
-                .first(where: { $0.id != content.currentUserId })
-
-            return (firstOtherMember?.imageURL, firstOtherMember?.isOnline ?? false)
-        }()
-
-        presenceAvatarView.avatarView.imageView.loadImage(
-            from: avatarURL,
-            placeholder: appearance.images.userAvatarPlaceholder4,
-            preferredSize: .avatarThumbnailSize,
-            components: components
-        )
-        presenceAvatarView.isOnlineIndicatorVisible = isOnlineIndicatorVisible
+        }
+        
+        group.notify(queue: DispatchQueue.main) {
+            completion(avatars, channelId)
+        }
+    }
+    
+    private func getCombinedAvatarImage(from avatars: [UIImage]) -> UIImage? {
+        var combinedImage: UIImage?
+        
+        let images = avatars.map { $0.scaled(to: .avatarThumbnailSize) }
+        
+        // The half of the width of the avatar
+        let halfContainerSize = CGSize(width: CGSize.avatarThumbnailSize.width / 2, height: CGSize.avatarThumbnailSize.height)
+        
+        if images.count == 1 {
+            combinedImage = images[0]
+        } else if images.count == 2 {
+            let leftImage = images[0].crop(to: halfContainerSize) ?? appearance.images.userAvatarPlaceholder1
+            let rightImage = images[1].crop(to: halfContainerSize) ?? appearance.images.userAvatarPlaceholder1
+            combinedImage = UIImageCombiner.combine(
+                images: [
+                    leftImage,
+                    rightImage
+                ],
+                orientation: .horizontal
+            )
+        } else if images.count == 3 {
+            let leftImage = images[0].crop(to: halfContainerSize)
+            let rightImage = UIImageCombiner.combine(
+                images: [
+                    images[1],
+                    images[2]
+                ],
+                orientation: .vertical
+            )?
+                .scaled(to: .avatarThumbnailSize)
+                .crop(to: halfContainerSize) ?? appearance.images.userAvatarPlaceholder3.crop(to: halfContainerSize)
+            combinedImage = UIImageCombiner.combine(
```

Comment:
Like mentioned before, we should not use singletons here, it is better that the `ImageLoader` and `ImageCombiner` are dependencies of `ChatChannelAvatarView` so that they can be easily changed.